packages/core/src/components/slider/_slider.scss

```diff
@@ -37,21 +37,48 @@ $track-height: $track-size !default;
   }
 }
 
-.pt-slider-track,
-.pt-slider-progress {
-  @include slider-track-orientation($handle-size, $track-size, $vertical: false);
-  position: absolute;
-  border-radius: $pt-border-radius;
+.pt-slider-track:first-child {
   background: rgba($gray1, 0.2);
 
   .pt-dark & {
     background: rgba($black, 0.5);
   }
 }
 
+.pt-slider-track,
+.pt-slider-progress {
+  @include slider-track-orientation($handle-size, $track-size, $vertical: false);
+  position: absolute;
+  border-radius: $pt-border-radius;
+}
+
 .pt-slider-progress,
 .pt-dark .pt-slider-progress {
-  background: $pt-intent-primary;
+  &.pt-lower {
+    border-top-left-radius: 0;
+    border-bottom-left-radius: 0;
+  }
+
+  &.pt-upper {
+    border-top-right-radius: 0;
+    border-bottom-right-radius: 0;
+  }
+
+  &.pt-intent-primary {
```

Comment:
I'm not a sass wizard, and I imagine that there's a much nicer way of doing these intent blocks. Any thoughts?

Reply:
you can loop through all the intents. check out how buttons, callouts, and other intent-supported components do it